ruby_event_store/lib/ruby_event_store/client.rb

```diff
@@ -29,10 +29,7 @@ def publish(events, stream_name: GLOBAL_STREAM, expected_version: :any)
       serialized_events = serialize_events(enriched_events)
       append_to_stream_serialized_events(serialized_events, stream_name: stream_name, expected_version: expected_version)
       enriched_events.zip(serialized_events) do |event, serialized_event|
-        with_metadata(
-          correlation_id: event.metadata[:correlation_id] || event.event_id,
-          causation_id:   event.event_id,
-        ) do
+        with_metadata(CorrelatedDecorator.metadata_for(event)) do
```

Comment:
this is a test of `with_metadata`, not of `CorrelatedDecorator`, let's not use it.

Reply:
@paneq right, that was a mistake